sourcegraph/sourcegraph
# Search: Add aggregation panel mock UI

Closes https://github.com/sourcegraph/sourcegraph/issues/39954
Closes https://github.com/sourcegraph/sourcegraph/issues/40088

[RFC](https://docs.google.com/document/d/1yLV7MGOEXYT3yPvBxHHm8y2-p737KIrju_QPqlhBYz4/edit#)
[Figma designs](https://www.figma.com/file/cKSeCtmBh9aiPsAVoKsLgM/Aggregation-insights%3A-display-%22grouped%22-insights-for-existing-searches-%2339126?node-id=275%3A1628)

## Background 
New aggregation UI on the search result page

| Standard UI  | Simple UI mode |
| ------------- | ------------- |
| <img width="1128" alt="Screenshot 2022-08-12 at 14 33 21" src="https://user-images.githubusercontent.com/18492575/184363454-5c370c3c-fe22-4956-8145-49710f48fa2d.png"> | <img width="1128" alt="Screenshot 2022-08-12 at 14 38 24" src="https://user-images.githubusercontent.com/18492575/184363565-03e01331-6561-4e79-b6be-05e392d5e740.png"> |

This PR adds mock (no data) UI for aggregation panel UI on the search result page. 

## For reviewers 
In order to make it change as simple as possible and put all parts of this UI in just one single file. Further steps around codebase quality will be addressed in follow issue about refactoring search-ui components API here https://github.com/sourcegraph/sourcegraph/issues/40285

## Follow-ups
- Refactor search UI components API in order to decouple UI and business logic in reusable search UI components https://github.com/sourcegraph/sourcegraph/issues/40285
- The sidebar panel is stored in the `search-ui` package. Currently, chart UI components are stored in the `web` package. Because of this, PR introduces a cyclic link since we import chart UI into search from `web` and the `web` package imports `search-ui` for the search result page. Webpack can handle this, and this is how it works at the moment, but we should move chart UI into Wildcard (or another separate package) to resolve this problem in our codebase https://github.com/sourcegraph/sourcegraph/issues/40136. _**Update: this issue has been resolved**_

## Test plan
- Go to the search result page 
- See that nothing is changed and the side panel layout looks correct for mobile and desktop 
- Turn on the global feature flag `search-aggregation-filters.`
- See that you can see mock UI for the aggregation panel, panel should look properly for mobile and desktop 



## App preview:

- [Web](https://sg-web-vk-add-aggregation-panel-mock-ui.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-tpflzxyoqe.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
